network/policy/policy_windows.go

```diff
@@ -343,6 +347,27 @@ func GetHcnPortMappingPolicy(policy Policy) (hcn.EndpointPolicy, error) {
 	return portMappingPolicy, nil
 }
 
+// GetHcnACLPolicy returns ACL policy.
+func GetHcnACLPolicy(policy Policy) (hcn.EndpointPolicy, error) {
+	aclEndpolicySetting := hcn.EndpointPolicy{
+		Type: hcn.ACL,
+	}
+
+	var aclPolicySetting hcn.AclPolicySetting
+	if err := json.Unmarshal(policy.Data, &aclPolicySetting); err != nil {
+		return aclEndpolicySetting, err
+	}
+
+	aclPolicySettingBytes, err := json.Marshal(aclPolicySetting)
```

Comment:
Why marshal into bytes when we already have the bytes in `policy.Data`?

Reply:
+1. Here you are unmarshalling and marshalling back. For the other policies such as route, we were unmarshalling, extracting the needed settings, add those in hcn policy settings and then marshalling to []byte.

Reply:
This is to make sure, that it meets ACPolicy construct, rather waiting until the endpoint creation creates.

Reply:
Let me know if you wait to until the endpoint creation.

Reply:
I think it's fine, but please add a comment to explain so that later someone doesn't come through cleaning this up and accidentally removes the intended behavior

Reply:
done.